packages/grafana-ui/src/components/PanelChrome/HoverWidget.tsx

```diff
@@ -0,0 +1,104 @@
+import { css } from '@emotion/css';
+import classnames from 'classnames';
+import React, { ReactElement, useCallback, useRef, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+import { Icon } from '../Icon/Icon';
+
+import { PanelMenu } from './PanelMenu';
+
+interface Props {
+  children?: React.ReactNode;
+  menu: ReactElement | (() => ReactElement);
+  title?: string;
+  offset?: number;
+  dragClass?: string;
+}
+
+export function HoverWidget({ menu, title, dragClass, children, offset = -32 }: Props) {
+  const styles = useStyles2(getStyles);
+  const draggableRef = useRef<HTMLDivElement>(null);
+
+  // Capture the pointer to keep the widget visible while dragging
+  const onPointerDown = useCallback((e: React.PointerEvent<HTMLDivElement>) => {
+    draggableRef.current?.setPointerCapture(e.pointerId);
+  }, []);
+
+  const onPointerUp = useCallback((e: React.PointerEvent<HTMLDivElement>) => {
+    draggableRef.current?.releasePointerCapture(e.pointerId);
+  }, []);
+
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  if (children === undefined || React.Children.count(children) === 0) {
+    return null;
+  }
+
+  return (
+    <div className={classnames(styles.container, { 'show-on-hover': !menuOpen })} style={{ top: `${offset}px` }}>
+      <div
+        className={classnames(styles.square, styles.draggable, dragClass)}
+        onPointerDown={onPointerDown}
+        onPointerUp={onPointerUp}
+        ref={draggableRef}
+      >
+        <Icon name="draggabledots" />
+      </div>
+      {children}
+      <div className={styles.square}>
+        <PanelMenu
```

Comment:
also here
the menu doesn't receive `dragClassCancel` and can be used for dragging